core/gesture.js

```diff
@@ -604,6 +604,7 @@ Blockly.Gesture.prototype.doFieldClick_ = function() {
  */
 Blockly.Gesture.prototype.doBlockClick_ = function() {
   // Block click in an autoclosing flyout.
+  if (this.targetBlock_.disabled) return;
```

Comment:
Block clicks on disabled blocks should still work everywhere else.  Returning early also means no call to `Blockly.Events.setGroup(false)`, which would be bad.  Do this instead:

```
  if (this.flyout_ && this.flyout_.autoClose) {
    if (!this.targetBlock_.disabled) {
      if (!Blockly.Events.getGroup()) {
        Blockly.Events.setGroup(true);
      }
      var newBlock = this.flyout_.createBlock(this.targetBlock_);
      newBlock.scheduleSnapAndBump();
    }
  }
```